src/main/scala/is/hail/variant/Genotype.scala

```diff
@@ -177,8 +178,56 @@ class RowGenotype(r: Row) extends Genotype {
 
 object Genotype {
   def buildGenotypeExtractor(t: Type): (Any) => Genotype = {
-    assert (t == TGenotype)
-    (a: Any) => a.asInstanceOf[Genotype]
+    if (t == TGenotype)
+      (a: Any) => a.asInstanceOf[Genotype]
+    else {
+      def queryOpt[T >: Null](field: String)(implicit hr: HailRep[T]): Option[(Any) => T] = {
+        try {
+          val (ft, fq0) = t.queryTyped(field)
+          if (ft == hr.typ)
+            Some((a: Any) => fq0(a).asInstanceOf[T])
+          else
+            None
```

Comment:
I assume this is guarding for the case where the field exists but someone has defined it as a different type than the one expected by the Genotype class.